calibration_assistant.cpp

```diff
@@ -0,0 +1,1250 @@
+    case STATE_CALIBRATING_SECONDARY:
+        m_calibrationActive = true;
+        break;
+    case STATE_CALIBRATED:
+    case STATE_GUIDING:
+    {
+        m_calibrationActive = false;
+        m_monitoringCalibration = false;
+        EvaluateCalibration();
+        m_pCalibrateBtn->Enable(true);
+        break;
+    }
+
+    }
+}
+
+void CalibrationAssistant::OnTimer(wxTimerEvent& evt)
+{
+    if (!m_monitoringCalibration)
+        UpdateCurrentPosition(true);
+    else
+        TrackCalibration(pFrame->pGuider->GetState());
+}
+
+void CalibrationAssistant::UpdateCurrentPosition(bool fromTimer)
+{
+    double ra, dec, lst;
+    double hourAngle;
+    const int INVALID_DEC = 100;
+    static double lastDec = INVALID_DEC;
+
+    if (!pPointingSource->GetCoordinates(&ra, &dec, &lst))
+    {
+        m_currentRA = ra;
+        m_currentDec = dec;
+        hourAngle = norm(lst - ra, -12.0, 12.0);
+        if (hourAngle < 0)
+            m_pCurrEast->SetValue(true);
+        else
+            m_pCurrWest->SetValue(true);
+        if (!m_isSlewing)
+        {
+            if (m_pCurrWest->GetValue() != m_pTargetWest->GetValue())
+                m_pWarning->SetLabelText(_("MERIDIAN FLIP!"));
+            else
+                m_pWarning->SetLabelText(wxEmptyString);
+        }
+        else
+            m_pWarning->SetLabelText(_("WATCH SCOPE DURING SLEWING TO INSURE SAFETY"));
+        m_pCurrOffset->SetValue(wxString::Format("%.1f", abs(hourAngle * 15.0)));
+        m_pCurrDec->SetValue(wxString::Format("%+.1f", dec));
+        if (!m_meridianFlipping)
+        {
+            if (fromTimer)
+            {
+                if (fabs(lastDec - dec) > 10)
+                {
+                    if (lastDec != INVALID_DEC)
+                        ShowExplanationMsg(dec);
+                    lastDec = dec;
+                }
+            }
+            else
+                ShowExplanationMsg(dec);
+        }
+    }
+    else
+    {
+        ShowError(_("Mount can't report its pointing position"), true);
+        if (m_pTimer)
+            m_pTimer->Stop();
+    }
+}
+// Return of 'true' means error occurred
+bool CalibrationAssistant::GetCalibPositionRecommendations(int* HA, int* Dec) const
+{
+    double hourAngle;
+    int bestDec;
+    int bestOffset;
+    double lst;
+    double ra;
+    double dec;
+    double eqAltitude;
+    bool pointingEast = false;
+    bool errorSeen = false;
+
+    bestDec = defBestDec;
+    bestOffset = defBestOffset;
+    try
+    {
+        if (pPointingSource && pPointingSource->CanReportPosition())
+        {
+            if (pPointingSource->PreparePositionInteractive())
+                return true;
+            if (pPointingSource->GetCoordinates(&ra, &dec, &lst))
+            {
+                throw ERROR_INFO("CalPositionRecommendations: Mount not reporting pointing position");
+            }
+            hourAngle = norm(lst - ra, -12.0, 12.0);
+            if (hourAngle <= 0)

```

Comment:
C/C++ can do nasty things with implicit conversions so I find it helpful to use explicitly-typed constants rather than letting the compiler do implicit casting. 
In this case that means
```suggestion
            if (hourAngle <= 0.)
```
likewise for 30 => 30. 90 => 90. below    (30. and 30.0 would both be ok -- both are the same double whereas 30 is int)